### ISSUE TITLE
json client, wrong content-type on delete method

### ISSUE THREAD

**Issue by spikhoff** [restify/node-restify#539]
Using latest restify (2.6.1) when json client makes http DELETE request it sets content-type to application/octet-stream instead application/json.

Check out this gist:
https://gist.github.com/spikhoff/9229833

**Comment by micahr** [comment]
This looks like an issue with the router. Not sure why you would be getting flagged as a "typed" request, as only PUT, POST, and PATCH requests should qualify. Add it to the next minor release milestone.

**Comment by retrohacker** [comment]
Thank you for the clear and concise repro case! Marking this as Help Wanted for now until a contributor frees up and gets a chance to tackle it. :heart:

Including the repro case here to save a click:

```javascript
var restify = require('restify');

// create server
var server = restify.createServer();

// Blocks your chain on reading and parsing the HTTP request body.
// Switches on Content-Type and does the appropriate logic.
// application/json, application/x-www-form-urlencoded and multipart/form-data are currently supported.
server.use(restify.bodyParser({rejectUnknown: true}));

server.del('/endpoint', function (req, res, next) {
  return next();
});

// create client
var client = restify.createJsonClient({
  version: '*',
  url: 'http://0.0.0.0:8080'
});


server.listen(8080, function () {
  // And here we get UnsupportedMediaTypeError when bodyParser rejectUnknown is set to true.
  // JsonClient sends application/octet-stream instead application/json.
  client.del('/endpoint', function(err) {
    console.log(err);
  });
});
```

**Comment by hekike** [comment]
Moved to:
https://github.com/restify/clients/issues/163

**Comment by einarjohnson** [comment]
Was this issue ever resolved? I'm seeing the same behaviour in 7.2.1. My client sends in a DELETE request and the bodyparser (with rejectUnknown: true) is throwing it back as a "Unsupported Media Type" error.
